LIBRARY.md

```diff
@@ -23,8 +23,10 @@ The library also provides some additional helper modules:
 ## Python Bindings
 Python bindings are provided for all modules in `python/`. Python bindings follow the same naming conventions as the C++ implementation, with the caveat that template types are realized by appending the type to the class name. For example given a C++ class `jrl::Class<Type>` the corresponding python binding will be `jrl.ClassType`. 
 
+Enable building the python bindings with `JRL_BUILD_PYTHON` CMake option. The bindings can be installed with `make jrl-python-install` and uninstalled with `jrl-python-uninstall`. These commands delegate to pip under the hood and can work with virtual environments by setting the CMake variable `PYTHON_EXECUTABLE` to the proper path for your environment (find by running `which python` when the environment is active). For this to work GTSAM must also be installed in the environment which can be done in a similar manner to the method described here for JRL.
```

Comment:
Hmm thats interesting, it is necessary for me to set. 
And you are correct setting the executable essentially defines which `pip` is used for the install.

Reply:
Follow up: Do you have a system python? or do you only use conda?
context: https://stackoverflow.com/questions/68302655/embedding-python-witn-pybind11-using-virtualenv

Reply:
I do have a system python, but rarely use it. 

I think `PYTHON_EXECUTABLE` should be set by cmake via whatever python cmake finds... when you set `PYTHON_EXECUTABLE` by hand is cmake finding your virtual env python? Or is it building against your system python?

Reply:
Okay I figured things out!!

`FindPython` (now explicitly called via `find_package(Python)`) will find the active python interpreter including ones from virtual environments if active.

However! this CMakeScript caches all variables and does not get re-run after you change python environments.

Therefore, if you build the library, then activate an environment and then try to install python, things wont work because it will be using the wrong python version. 

To address this issue I added some documentation.